documentation/user_docs/docs/manual/Advanced_Multifit.rst

```diff
@@ -2,133 +2,160 @@
+something Gaussian (whoo! Go C.F. Gauss, I guess).
 
-If multifit is defined as a method of the class, then one can use the capability of nesting functions within the method
-to accept different fit function syntax. This is done, for example, for sqw objects when the fit functions to multifit,
-and equivalently multifit_func, are just a 1D/2D...4D Gaussian (according to the dimensionality of the sqw object).
+If ``multifit`` is defined as a method of the class, then one can use the
+capability of nesting functions within the method to accept different fit
+function syntax. This is done, for example, for ``sqw`` objects when the fit
+functions to ``multifit``, and equivalently ``multifit_func``, are just an n-D
+Gaussian (according to the dimensionality of the ``sqw`` object).
 
 
 Utility methods
 ---------------
 
-These are required to enable multifit to work with objects
+These are methods ``multifit`` can use to fit user-defined objects
+
+mask
+~~~~
+
+.. code-block:: matlab
+
+   wout = mask(win, msk)
+
+
+A method that masks data points from further calculation. The output object must
+be a valid instance of the class in which the masked values have been removed in
+whatever sense the class requires.
+
+.. _mask_points:
 
-wout = mask(win, msk)
+mask_points [optional]
 ~~~~~~~~~~~~~~~~~~~~~~
 
-A method that masks data points from further calculation. The output object must be a valid instance of the class in
-which the masked values have been removed in whatever sense the class requires.
+.. code-block:: matlab
 
+   [msk, ok, mess] = mask_points(win, 'keep', xkeep, 'remove', xremove, 'mask', msk_in)
 
-[y,var,msk] = sigvar_get(win)
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Create a mask array given ranges of x-coordinates to keep, remove or mask from the array. The elements of a mask array are
+``true`` for those data points which are to be retained.
+
+Function must output a logical flag ``ok``, with message string if ``ok==false`` rather than terminate.
 
-A method that returns the intensity and variance arrays from the objects, along with a mask array that indicates which
-elements are to be retained (where elements of msk are true, the corresponding elements of y and var are retained). The
-output arrays y and var must have the same size and shape; msk must have the same number of elements (but can be a
-different shape). The array msk must be understood by the method 'mask' defined below.
+(It is possible to have the function terminate if ``ok`` and ``mess`` are not given as return arguments; it is the
+advanced syntax that is required within ``multifit``).
+
+sigvar_get
+~~~~~~~~~~
 
+.. code-block:: matlab
 
-wsum = w1 + w2
-~~~~~~~~~~~~~~
+   [y, var, msk] = sigvar_get(win)
 
-If a background function is provided, addition of objects must be defined (requires overloading of the addition
-operator with a method named plus.m).
+A method that returns the intensity and variance arrays from the objects, along
```

Comment:
I'm presuming that this is a set of instructions to the developer to implement sigvar_get, but it reads a bit as if this is the description of what the provided sigvar_get does. In particular it is unclear how sigvar_get will know what to mask. Some orientation for the reader would be useful here.